### PR TITLE
Add Agents Support to SDK

### PR DESCRIPTION
**Pull request by tmichaeldb**
Fixes RAG-26.

Depends on changes in [#9067](https://github.com/mindsdb/mindsdb/pull/9067)

This PR adds support for creating & using agents from underlying MindsDB Langchain models. Currently, only text-to-SQL is supported. A follow-up PR will add a retrieval skill for RAG functionality.

See `/examples/using_agents.py` for example usage:
```python
import mindsdb_sdk

con = mindsdb_sdk.connect()

# We currently support Langchain as a backend.
print('Creating underlying langchain model for the agent to use...')
try:
    langchain_engine = con.ml_engines.get('langchain')
except Exception:
    # Create the engine if it doesn't exist.
    langchain_engine = con.ml_engines.create('langchain', handler='langchain')

# Actually create the underlying model the agent will use.
langchain_model = con.models.create(
    'agent_model',
    predict='answer',
    engine='langchain',
    prompt_template='You are a spicy, cheeky assistant. Add some personality and flare when responding to the user question: {{question}}',
    model_name='gpt-4-0125-preview' # This is the underlying LLM. Can use OpenAI, Claude, local Ollama, etc
    # Can optionally set LLM args here. For example:
    # temperature=0.0,
    # max_tokens=1000,
    # top_p=1.0,
    # top_k=0,
    # ...
)
print('Agent ready to use.')

# Now create an agent that will use the model we just created.
agent = con.agents.create('new_agent', langchain_model)
print('Ask a question: ')
question = input()
answer = agent.completion([{'question': question, 'answer': None}])
print(answer.content)
```

### CODE REVIEW

**Comment by ea-rus** [comment]
I've merged #97. Now if you update this branch from staging - CI should be fine

File: mindsdb_sdk/agents.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        self.collection = collection
+
+    def completion(self, messages: List[dict]) -> AgentCompletion:
+        return self.collection.completion(self.name, messages)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(name: {self.name})'
+
+    def __eq__(self, other):
+        if self.name != other.name:
```
Comment (ea-rus) [inline mindsdb_sdk/agents.py:46]: how this method should be used. in which use-case user compares this objects? 
if it is required: from mindsdb server perspective two agents can't have the same name in one project. So name-project_name is already unique identifier.

File: mindsdb_sdk/agents.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            [Skill.from_json(skill) for skill in json['skills']],
+            json['params'],
+            json['created_at'],
+            json['updated_at'],
+            collection
+        )
+
+
+class Agents(CollectionBase):
+    """Collection for agents"""
```
Comment (ea-rus) [inline mindsdb_sdk/agents.py:72]: He have documentation [autogenerated](https://mindsdb.github.io/mindsdb_python_sdk/) from docstrings. 
Would be great to have here (and in Skills class) some brief info how to use it. 
Examples from other classes [1](https://github.com/mindsdb/mindsdb_python_sdk/blob/628a7c7d1494ac7e045472a4e08ad90eafd2ed09/mindsdb_sdk/views.py#L38), [2](https://github.com/mindsdb/mindsdb_python_sdk/blob/06202aea36d67823c53b16f979e40dfd711e5439/mindsdb_sdk/tables.py#L179)

File: tests/test_sdk.py [hunk 3]
```diff
@@ -1069,3 +1072,351 @@ def check_project_jobs(self, project, mock_post):
...
+            'updated_at': updated_at
+        }
+        response_mock(mock_post, data)
+
+        # Create the agent.
+        server = mindsdb_sdk.connect()
+        new_agent = server.agents.create(
+            name='test_agent',
+            model=Model(None, {'name':'m1'}),
+            skills=[Skill('test_skill', 'test', {'k1': 'v1'})],
```
Comment (ea-rus) [inline tests/test_sdk.py:1168]: Maybe to forbid to use generic Skill and force the user to use specific skill: SQLSkill or KBSkill ?
It helps to send only required params.

**Review by ea-rus — CHANGES_REQUESTED** [review]

**Comment by ea-rus** [comment]
I've added some fixes for tests and docs, CI works now

File: tests/test_sdk.py (hunk 3 above)
Comment (ea-rus) [reply tests/test_sdk.py]: For now (when we don't have all wrappers for all skills) we could keep to use just generic Skill class

File: mindsdb_sdk/agents.py (hunk 1 above)
Comment (ea-rus) [reply mindsdb_sdk/agents.py:81]: The main idea: skill and agent can be identified by name (and project), we can't have the same name in one project. So we don't need to compare all columns to check if it is the same object in mindsdb. 
If the purpose is only for tests (to compare result with expectation) we could move this comparator to tests (also we can do it later)

**Review by dusvyat — APPROVED** [review]
nice looks good to me

File: mindsdb_sdk/agents.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            name,
+            updated_agent.name,
+            updated_agent.model_name,
+            list(skills_to_add),
+            list(skills_to_remove),
+            updated_agent.params
+        )
+        return Agent.from_json(data, self)
+
+    def delete(self, name: str):
```
Comment (ea-rus) [inline mindsdb_sdk/agents.py:217]: could we rename this method to 'drop' for consistency with methods for other entities in SDK API (and for skills too) ?

**Review by ea-rus — APPROVED** [review]

**Pull request merged**